packages/next/src/server/app-render/create-component-tree.tsx

```diff
@@ -574,6 +575,7 @@ async function createComponentTreeInternal({
       // If we are passing searchParams to a server component Page we need to track their usage in case
       // the current render mode tracks dynamic API usage.
       props.searchParams = createDynamicallyTrackedSearchParams(query)
+      props.params = createDynamicallyTrackedParams(props.params)
```

Comment:
I think we should do this on line 555 when we initialize `props.params` as we wouldn't want a branch to not get the tracked params.